urllib3/contrib/pyopenssl.py

```diff
@@ -204,9 +204,11 @@ def _send_until_done(self, data):
                 continue
 
     def sendall(self, data):
-        while len(data):
-            sent = self._send_until_done(data)
-            data = data[sent:]
+        total_sent = 0
+        while total_sent < len(data):
+            # OpenSSL will only write 16K at once
+            sent = self._send_until_done(data[total_sent:total_sent+16384])
```

Comment:
+1 that would be better, though I generally accept good comments as substitutes for named vars. :)